cms/navigation/models.py

```diff
@@ -0,0 +1,132 @@
+from typing import ClassVar
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from wagtail.admin.panels import FieldPanel
+from wagtail.blocks import CharBlock, ListBlock, PageChooserBlock, StructBlock, URLBlock
+from wagtail.contrib.settings.models import BaseSiteSetting, register_setting
+from wagtail.snippets.models import register_snippet
+
+from cms.core.fields import StreamField
+
+
+# Custom LinkBlock to support both pages and URLs
+class BaseLinkBlock(StructBlock):
+    page = PageChooserBlock(required=False)
+    url = URLBlock(required=False, label="URL")
+    title = CharBlock(required=False, help_text="Optional. Displayed as the link text.")
+
+    def clean(self, value: dict) -> dict:
+        value = super().clean(value)
+        if not value.get("page") and not value.get("url"):
+            raise ValidationError("Either a page or a URL must be provided.")
+        return value
+
+    class Meta:
+        abstract = True
+
+
+class LinkBlock(BaseLinkBlock):
+    class Meta:
+        icon = "link"
+        label = "Link"
+
+
+class ThemeLinkBlock(BaseLinkBlock):
+    page = PageChooserBlock(required=False, page_type="themes.ThemePage")
+
+    class Meta:
+        icon = "link"
+        label = "Theme Link"
+
+
+class TopicLinkBlock(BaseLinkBlock):
+    page = PageChooserBlock(required=False, page_type="topics.TopicPage")
+
+    class Meta:
+        icon = "link"
+        label = "Topic Link"
+
+
+class HighlightsBlock(BaseLinkBlock):
+    description = CharBlock(required=True, max_length=50, help_text="E.g., It's never been more important")
```

Comment:
Suggestion: use 'View our latest and upcoming releases.' as the example help text as this is perhaps more meaningful to an editor, whereas the current example is very specific to a survey link. (Note that this example includes a full stop as Amelia confirmed these descriptions need a full stop for the prototype)

Reply:
Done

Reply:
Done =)